### DISCUSSION TITLE
물리적 서버 1대당 브로커의 적절한 갯수는?

### DISCUSSION THREAD

**Discussion by bonyeon0605** [kyungjunleeme/kafka_study#10] [Q&A] [answered]
물리적 서버 1대당 브로커 1개를 사용하면 좋겠지만(브로커 하나가 가용할 수 있는 자원이 많기 때문에)
실제 현업에서는 어떻게 구성되어 있을까? 물리적 서버 1대당 브로커 2,3개?

+ 쿠버네티스 클러스터 관점도 추가 필요

**Comment by hellices** [comment, marked as answer]
kafka official document에서는 특별한 가이드가 없고,
[confluent](https://docs.confluent.io/platform/current/installation/system-requirements.html?utm_source=chatgpt.com)에서 제공하는 가이드에서 on-premise에서 설치 시 권장하는 사양을 찾아볼 수 있습니다.

다만 권장스펙의 요구사항이 굉장히 높고 저 사양을 갖추고 작은 규모로 시작하는 방식은 보드진에게 설득되지 않을 가능성이 높다고 봅니다.
저희는 4 core / 8GB 3대의 브로커로 시작했고, 현재 일부 DB 동기화에서 사용하고 있습니다. 이 정도 규모에서도 카프카 자체의 성능은 큰 이슈가 없으며, 앞에 설치한 kafka connect 등의 producer / consumer의 성능이 실제 동기화 지연에 주요 원인으로 파악하고 있습니다.

쿠버네티스에서는 동적으로 pod가 배포되고, worker node가 몇 대로 구성되는지에 좌우되는게 더 많으며, 설치 자체는 큰 차이가 없다고 봅니다.

**Comment by kyungjunleeme** [comment]
![image](https://github.com/user-attachments/assets/183dbed9-3778-4316-99ac-36cf66f96143)

![image](https://github.com/user-attachments/assets/e741ed8d-c129-4268-987a-501899c8d51f)

https://youtu.be/XyuqoWUCdGA?si=jLjZFaWeob58HU0o

8분 16초 내용 보충

**Comment by kyungjunleeme** [comment]
하드웨어 스펙 요구사항
https://docs.confluent.io/platform/current/installation/system-requirements.html
